docker/nginx.tmpl

```diff
@@ -278,6 +278,15 @@ server {
     proxy_set_header Host asset-manager.dev.gov.uk;
     proxy_pass {{ trim $proto }}://{{ trim $upstream_name }};
   }
+  location ~ /cloud-storage-proxy/(.*) {
+    internal;
+    set $download_url $1$is_args$args;
+    proxy_pass $download_url;
+  }
```

Comment:
Any chance you could explain a little the purpose of this? I don't know nginx well enough to understand exactly what it's doing

Reply:
This replicates the critical bits of the [`cloud-storage-proxy` Nginx location definition](https://github.com/alphagov/govuk-puppet/blob/dd42edb08b978feb95627032211a1084260be2cb/modules/govuk/manifests/apps/asset_manager.pp#L131).

The Asset Manager Rails app [sets the `X-Accel-Redirect` response header](https://github.com/alphagov/asset-manager/blob/master/app/controllers/base_media_controller.rb#L18-L19). The value of this header is a path prefixed by `/cloud-storage-proxy/` followed by an "argument" which is the S3 URL.

When Nginx receives this response it does an internal redirect within the virtual host and matches the `cloud-storage-proxy` location block above. This causes Nginx to extract the S3 URL (`$download_url`) and proxy the request to that S3 URL.

Does that make sense? I'm happy to add some comments to the code to explain if you think that would be useful.

Reply:
Yeah that makes sense thanks. I think a comment would be useful (even if it's just a link to the file it's replicating.

Reply:
I've added a comment in 25bcbe80e56f360458f22c4dfffc966a0d225b13.